contracts/tec/contracts/src/MixinTECApprovalVerifier.sol

```diff
@@ -0,0 +1,199 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.3;
+pragma experimental "ABIEncoderV2";
+
+import "@0x/contracts-exchange-libs/contracts/src/LibExchangeSelectors.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibOrder.sol";
+import "@0x/contracts-utils/contracts/src/LibBytes.sol";
+import "@0x/contracts-utils/contracts/src/LibAddressArray.sol";
+import "./libs/LibTECApproval.sol";
+import "./libs/LibZeroExTransaction.sol";
+import "./mixins/MSignatureValidator.sol";
+import "./mixins/MTECApprovalVerifier.sol";
+
+
+// solhint-disable avoid-tx-origin
+contract MixinTECApprovalVerifier is
+    LibExchangeSelectors,
+    LibTECApproval,
+    LibZeroExTransaction,
+    MSignatureValidator,
+    MTECApprovalVerifier
+{
+    using LibBytes for bytes;
+    using LibAddressArray for address[];
+
+    /// @dev Validates that the 0x transaction has been approved by all of the feeRecipients
+    ///      that correspond to each order in the transaction's Exchange calldata.
+    /// @param transaction 0x transaction containing salt, signerAddress, and data.
+    /// @param transactionSignature Proof that the transaction has been signed by the signer.
+    /// @param approvalExpirationTimeSeconds Array of expiration times in seconds for which each corresponding approval signature expires.
+    /// @param approvalSignatures Array of signatures that correspond to the feeRecipients of each order in the transaction's Exchange calldata.
+    function assertValidTECApprovals(
+        LibZeroExTransaction.ZeroExTransaction memory transaction,
+        bytes memory transactionSignature,
+        uint256[] memory approvalExpirationTimeSeconds,
+        bytes[] memory approvalSignatures
+    )
+        public
+        view
+    {
+        // Get the orders from the the Exchange calldata in the 0x transaction
+        LibOrder.Order[] memory orders = decodeFillDataOrders(transaction.data);
+
+        // No approval is required for non-fill methods
+        if (orders.length > 0) {
```

Comment:
This will catch `batchCancelOrders` requests. We need to special-case this somehow.

Reply:
Nvm, misread.